wherehows-web/app/components/search/containers/search-facets.ts

```diff
@@ -0,0 +1,162 @@
+  didInsertElement(this: SearchFacetsContainer) {
+    get(this, 'getPlatformsTask').perform();
+  }
+
+  /**
+   * The current source to narrow search results to
+   * @type {DatasetPlatform}
+   */
+  currentSource: string;
+
+  /**
+   * Lists data platforms available to restrict search results by source
+   * @type {Array<DatasetPlatform>}
+   * @private
+   */
+  _sources: Array<DatasetPlatform> = [];
+
+  /**
+   * Current state of selections of facets
+   * EI:
+   * {
+   *  source: {
+   *    hdfs: true
+   *  },
+   *  fabric: {
+   *    corp: true,
+   *    prod: true
+   *  }
+   * }
+   */
+  selections: IFacetsSelectionsMap;
+
+  /**
+   * Counts for the facets in a similar fashion of selections
+   */
+  counts: IFacetsCounts;
+
+  /**
+   * Gets the available platforms and extracts a list of dataset sources
+   * @type {(Task<Promise<Array<IDataPlatform>>, (a?: any) => TaskInstance<Promise<Array<IDataPlatform>>>>)}
+   */
+  getPlatformsTask = task(function*(this: SearchFacetsContainer): IterableIterator<Promise<Array<IDataPlatform>>> {
+    const platforms: Array<IDataPlatform> = yield readPlatforms();
+    const getDatasetPlatform = ({ name }: IDataPlatform): DatasetPlatform => name;
+    const dataPlatforms = arrayMap(getDatasetPlatform)(platforms);
+
+    get(this, '_sources').setObjects(dataPlatforms);
+  });
+
+  /**
+   * I will convert a string into a facet option with counts
+   * @param facetValue
+   * @param facetName
+   */
+  stringToFacetOption(facetValue: string): ISearchFacetOption {
+    return {
+      value: facetValue,
+      label: capitalize(facetValue)
+    };
+  }
+
+  /**
+   * Creates a list of options with radio props for the data platforms that can be selected as a search filter
+   * @type {(ComputedProperty<Array<ISearchSourceOption>>}
+   */
+  @computed('_sources.[]', 'counts')
+  get sources(this: SearchFacetsContainer): Array<ISearchFacetOption> {
+    return this._sources.map(source => this.stringToFacetOption(source));
+  }
+
+  /**
+   * Facets that are available right now.
+   * In the future, it should be fetched from the backend
+   */
+  @computed('sources', 'counts')
+  get facets(): Array<ISearchFacet> {
+    return [
+      {
+        name: 'fabric',
+        displayName: 'Fabrics',
+        values: ['prod', 'corp', 'ei', 'dev'].map(fabric => this.stringToFacetOption(fabric))
+      },
+      {
+        name: 'source',
+        displayName: 'Data Platform',
+        values: this.sources
+      }
+    ];
+  }
+
+  /**
+   * External closure action that triggers when facet changes
+   * @param _ Facet Selections
+   */
+  onFacetsChange(_: IFacetsSelectionsMap) {
+    //nothing
+  }
+
+  /**
+   * Internal action triggered when facet changes. It will update
+   * the state of selections in a redux fashion.
+   * @param facet The facet that changed
+   * @param facetValue the option of the facet that changed
+   */
+  onFacetChange(facet: ISearchFacet, facetValue: ISearchFacetOption) {
```

Comment:
I'm confused. Is `onFacetsChange` an external action, or is it multiple `onFacetChange` calls?

Reply:
When a facet changes, it triggers onFacetChange. That will update the state in the container component, but we need to change routes, so it triggers onFacetsChange which is the external interface of this container component. I understand that the naming can be confusing, but not sure what name to choose that helps understanding this.